--- old/src/share/vm/classfile/systemDictionary.hpp	2014-05-14 19:46:42.931244400 +0400
+++ new/src/share/vm/classfile/systemDictionary.hpp	2014-05-14 19:46:42.621133600 +0400
@@ -392,7 +392,7 @@
     return k;
   }
   static Klass* check_klass_Opt_Only_JDK14NewRef(Klass* k) {
-    assert(JDK_Version::is_gte_jdk14x_version() && UseNewReflection, "JDK 1.4 only");
+    assert(JDK_Version::is_gte_jdk14x_version(), "JDK 1.4 only");
     // despite the optional loading, if you use this it must be present:
     return check_klass(k);
   }
