--- old/src/share/vm/opto/library_call.cpp	2014-05-14 19:46:49.394328000 +0400
+++ new/src/share/vm/opto/library_call.cpp	2014-05-14 19:46:49.079214600 +0400
@@ -420,7 +420,6 @@
     return NULL;
 
   case vmIntrinsics::_getCallerClass:
-    if (!UseNewReflection)  return NULL;
     if (!InlineReflectionGetCallerClass)  return NULL;
     if (SystemDictionary::reflect_CallerSensitive_klass() == NULL)  return NULL;
     break;
