--- old/src/share/vm/runtime/globals.hpp	2014-05-12 15:11:15.189754600 +0400
+++ new/src/share/vm/runtime/globals.hpp	2014-05-12 15:11:14.850813600 +0400
@@ -3654,22 +3654,12 @@
                                                                             \
   /* New JDK 1.4 reflection implementation */                               \
                                                                             \
-  develop(bool, UseNewReflection, true,                                     \
-          "Temporary flag for transition to reflection based on dynamic "   \
-          "bytecode generation in 1.4; can no longer be turned off in 1.4 " \
-          "JDK, and is unneeded in 1.3 JDK, but marks most places VM "      \
-          "changes were needed")                                            \
-                                                                            \
   develop(bool, VerifyReflectionBytecodes, false,                           \
           "Force verification of 1.4 reflection bytecodes. Does not work "  \
           "in situations like that described in 4486457 or for "            \
           "constructors generated for serialization, so can not be enabled "\
           "in product.")                                                    \
                                                                             \
-  product(bool, ReflectionWrapResolutionErrors, true,                       \
-          "Temporary flag for transition to AbstractMethodError wrapped "   \
-          "in InvocationTargetException. See 6531596")                      \
-                                                                            \
   develop(intx, FastSuperclassLimit, 8,                                     \
           "Depth of hardwired instanceof accelerator array")                \
                                                                             \
